homeassistant/components/switch/wink.py

```diff
@@ -5,17 +5,110 @@
 https://home-assistant.io/components/switch.wink/
 """
 import asyncio
+import logging
+from os import path
+
+import voluptuous as vol
 
 from homeassistant.components.wink import WinkDevice, DOMAIN
 from homeassistant.helpers.entity import ToggleEntity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.config import load_yaml_config_file
 
 DEPENDENCIES = ['wink']
 
+_LOGGER = logging.getLogger(__name__)
+
+SERVICE_SET_CHIME_VOLUME = "wink_set_chime_volume"
```

Comment:
Why is this a switch? This should not be a switch.

Reply:
Switch is not a catch all component for anything that has a turn on/off state. It's for actual switches.

Please extract this device. For now you can put it under the Wink component or something?

Anything in the UI that has a turn_on and turn_off service will show a toggle, so it's the same UI experience.

Reply:
So the original Wink support siren was made by GoControl and it only supported on and off. That is why it was added as a switch. I am not sure what you mean by "It's for actual switches." 

So the reason this new siren was added as a switch is because python-wink's get_sirens() returns both types of sirens, the original on/off only and this new one which has much more functionality. 

I know there are some users requesting HA integration via the zwave component for this new Siren as well (None Wink users) Do you think a new component needs created? It really isn't something that I feel needs it's own UI component because it isn't something that the user would manually trigger. It is only used for notifications of some event.

Reply:
You don't need any own UI component. Any domain that has a turn_on and a turn_off service will automatically show a toggle.

Reply:
Agreed, that UI comment was more of a statement, I don't think it is needed. But there is still a question, should a new siren, or chime, or hardware based notifier component be created for these type of devices?

I know of four devices under this category 

http://www.gocontrol.com/detail.php?productId=13

https://domeha.com/z-wave-siren-sensor

https://aeotec.com/z-wave-doorbell

https://aeotec.com/z-wave-siren

All of these have different capabilities however.

Reply:
I rather not standardize these just yet, there are too little and they are all over the place. Let's just put it under the Wink component for now?

Reply:
Sounds good.